tests/src/Interop/StructMarshalling/PInvoke/MarshalStructAsLayoutSeq.cs

```diff
@@ -316,6 +317,13 @@ public static int Main()
     [DllImport("MarshalStructAsParam")]
     static extern bool MarshalStructAsParam_AsSeqByValSequentialAggregateSequentialWrapper(AggregateSequentialWrapper wrapper);
 
+    [DllImport("MarshalStructAsParam")]
+    static extern bool MarshalStructAsParam_AsSeqByValFixedBufferClassificationTest(FixedBufferClassificationTest str, float f);
+    [DllImport("MarshalStructAsParam")]
+    static extern bool MarshalStructAsParam_AsSeqByValFixedBufferClassificationTest(FixedBufferClassificationTestBlittable str, float f);
+    [DllImport("MarshalStructAsParam")]
+    static extern bool MarshalStructAsParam_AsSeqByValFixedBufferClassificationTest(FixedArrayClassificationTest str, float f);
+
     [DllImport("MarshalStructAsParam")]
```

Comment:
We should have more variants of HFAs. E.g. the `ExplicitHFA` like this should have the same ABI as the `HFA` - across all platforms. Do you agree? I do not think it is the case today.

```
[StructLayout(LayoutKind.Explicit, Size = 16)]
struct ExplicitHFA
{
    [Offset(0)]
    float f1;
    [Offset(4)]
    float f2;
    [Offset(8)]
    fixed float f[2];
}
```

Reply:
> how the HFA stuff works

It is simpler version of the SystemV calling convention - it just deals with homegenous floating point values. E.g. consider this:

```
[StructLayout(LayoutKind.Explicit, Size = 8)]
struct MyHFA
{
    [Offset(0)]
    float f1;
    [Offset(4)]
    float f2;
}
```

And matching C/C++ structure like this:

```
struct MyHFA
{
    float f1;
    float f2;
}
```

Today, it does not work on either Unix x64 or any ARM platform. Your fix makes it work on Unix x64. I think we should make it work on ARM/ARM64 as well for consistency.